.github/workflows/lint.yml

```diff
@@ -328,7 +328,7 @@ jobs:
           # /torch/csrc/generic/*.cpp is excluded because those files aren't actually built.
           # deploy/interpreter files are excluded due to using macros and other techniquies
           # that are not easily converted to accepted c++
-          python3 tools/linter/clang_tidy.py \
+          LANG=C.UTF-8 python3 tools/linter/clang_tidy.py \
```

Comment:
I haven't seen this flag before, can you point to some docs / an example that shows why we need it?

Reply:
We run into this in our forked branch.

https://github.com/csarofeen/pytorch/runs/2908634788
```UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 1422199: ordinal not in range(128)```

This is a result from some weird symbol we have in the comment. (a python open file which contains the special symbol throws this error)
```
//! Julien Herrmann, Yusuf Özkaya, Bora Uçar, Kamer Kaya, Umit Catalyurek.
``` 

I tried it on our CI machine and fixed it following this: https://katcipis.github.io/blog/python3-utf8-locale/
(I believe there's some weirdness in our CI machine setup, feel free to yell at me if there's a better way to fix this)

Reply:
I tried repo-ing locally by adding that to a file and running clang-tidy on it but it appears to work fine, we shuffled this code around it bit lately so we might have changed something.

If you're still hitting the errors after rebasing we can hardcode the encoding to utf-8, but I think it'd be better if we localize it as much as possible rather than use environment variables (so basically change wherever is erroring in `clang_tidy.py` in the `f.read()` to open `f` with `open(..., encoding="utf-8")`)

Reply:
No this doesn't repro on my local machine neither but only in the CI machine. I'll rebase and try that later tonight then. If the error persists I'll update it to specify encoding in `open` instead.

Reply:
Ha, looks like we updated from python3.6 to python3.8 in our container. So this is no longer needed. I'll remove this then.

Reply:
> looks like we updated from python3.6 to python3.8

I'm not entirely sure if that's the root cause.. But at least it doesn't repro on latest CI container. I can repro it from the container I pulled from my first commit in this PR: https://github.com/pytorch/pytorch/runs/2910690422